synapse/storage/util/id_generators.py

```diff
@@ -273,6 +273,19 @@ def _load_current_ids(
 
         # Load the current positions of all writers for the stream.
         if self._writers:
+            # We delete any stale entries in the positions table. This is
+            # important if we add back a writer after a long time; we want to
+            # consider that a "new" writer, rather than using the old stale
+            # entry here.
```

Comment:
Do we still need to the check on line 300 `if instance in self._writers`? 😄 

I don't think it can hurt, especially since the `DELETE` and `SELECT` queries are in different transactions?

Reply:
Technically we don't, no. The `SELECT` and `DELETE` are in the same transaction.

Not sure if its clearer to leave it or remove the check tbh.

Reply:
🤷 The logic becomes simplified a bit, but you then need to combine the two SQL statements to understand exactly what that dictionary is made up of.